File: src/contests/cg/puzzle/scala/rubik.scala
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

object Solution {
  def main(args: Array[String]): Unit = {
    val n = scala.io.StdIn.readLine().trim.toLong
    // Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
    val inner = if (n >= 2) n - 2 else 0L
    println(n * n * n - inner * inner * inner)
  }
}

File: src/contests/cg/puzzle/objectivec/rubik.m
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

#import <Foundation/Foundation.h>

int main() {
    long long n;
    scanf("%lld", &n);
    // Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
    long long inner = n >= 2 ? n - 2 : 0;
    printf("%lld\n", n * n * n - inner * inner * inner);
    return 0;
}
